lib/Serialization/DeserializeSIL.cpp

```diff
@@ -598,6 +625,19 @@ SILFunction *SILDeserializer::readSILFunction(DeclID FID,
   if (fn->empty() && errorIfEmptyBody)
     return nullptr;
 
+  // Check that there are no unresolved forward definitions of opened
+  // archetypes.
+  for (auto &KV : OpenedArchetypesTracker.getOpenedArchetypeDefs()) {
+    if (isa<GlobalAddrInst>(KV.getSecond())) {
+      llvm::dbgs() << "The following opened archetype is not resolved: " << KV.getFirst() << "\n";
```

Comment:
I would suggest using llvm::errs() here instead of llvm::dbgs(). I think you can probably pass in llvm::errs() to dump as well, no?

Also, I have not looked at what the type of auto is here, but it would be good to print out here the instruction that introduces the archetype as well so it is easy to find it.